tools/nebula-flink/src/main/java/org.apache.flink/connector/nebula/connection/client/AbstractNebulaClient.java

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package org.apache.flink.connector.nebula.connection.client;
+
+
+import com.facebook.thrift.TException;
+import com.vesoft.nebula.Client;
+import org.apache.flink.connector.nebula.utils.NebulaConstant;
+
+public abstract class AbstractNebulaClient {
+    protected int timeout = NebulaConstant.DEFAULT_TIMEOUT_MS;
+    protected int connectTimeout = NebulaConstant.DEFAULT_CONNECT_TIMEOUT_MS;
+    protected int connectionRetry = NebulaConstant.DEFAULT_CONNECT_RETRY;
+    protected  int executionRetry = NebulaConstant.DEFAULT_EXECUTION_RETRY;
+
+    public AbstractNebulaClient(){}
+
+    public AbstractNebulaClient(int timeout,int connectionRetry,int executionRetry){
+        this.timeout = timeout;
+        this.connectionRetry = connectionRetry;
+        this.executionRetry = executionRetry;
+    }
+
+    abstract public Client connectClient(String address, String username, String password) throws TException;
```

Comment:
why don't use HostAddress as parameter?

Reply:
The client accept List<HostAndPort> parameter, I convert string address to List<HostAndPort> in connectClient() function. Is HostAddress better?

Reply:
maybe because I found you will cover to Host Address for each client